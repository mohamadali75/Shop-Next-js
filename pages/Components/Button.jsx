import React from "react";
import { css } from "@emotion/css";
import { useTheme } from "@mui/material";



function Button({children,onClick}) {
    const theme = useTheme()
    return(
        <button className={css`
        border-radius: 20px;
       
        margin-bottom:10px;
         display: inline-block;
         padding: 8px;
          color: white;
          background-color: ${theme.palette.primary.main};
          text-align: center;
         cursor: pointer;
         &:hover {
               color: #555;
   }
   }
 `} type="button" onClick={onClick}>{children}</button>
    )
}
export default Button
