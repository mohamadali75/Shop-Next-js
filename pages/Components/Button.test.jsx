import renderer from 'react-test-renderer';
import React from "react"
import {cleanup, fireEvent, render} from '@testing-library/react';
import Button from './Button';

it('Button test', () => {
  const component = renderer.create(
    <Button>test</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();})
 
 
  afterEach(cleanup);

  it('Dom onClick Test', () => {
    let value=1
    const {getByText} = render(
      <Button onClick={()=>{value+=1}}>plus</Button>,
    );

  
    fireEvent.click(getByText(/plus/));
  
    expect(value).toBe(2)
  });
  


