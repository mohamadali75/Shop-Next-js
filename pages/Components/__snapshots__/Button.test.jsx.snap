// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test 1`] = `
<button
  className="css-xgfb6h"
  type="button"
>
  test
</button>
`;
